tidy3d/components/base.py

```diff
@@ -622,10 +623,24 @@ def _json_string(self, include_unset: bool = True, include_data: bool = True) ->
         """
         exclude_unset = not include_unset
 
-        # if not include_data, temporarily set the xr.DataArray encoder to return None
+        # if not data_file, temporarily set the xr.DataArray encoder to return None
         original_encoder = self.__config__.json_encoders[xr.DataArray]
-        if not include_data:
-            self.__config__.json_encoders[xr.DataArray] = lambda x: None
+        if data_file:
+            # Create/overwrite data file
+            with h5py.File(data_file, "w"):
```

Comment:
a little confused, what does this do? is it possible we dont need it and can just change the mode ("a") in the below function? For reference: https://docs.h5py.org/en/stable/high/file.html

Reply:
Yeah so this is in case the file already exists; on the other hand the function below is called for *every* DataArray found in the simulation, so we can't change the mode there because it will erase previous writes.

However I think it should be possible to do with a single file opening statement, I'll give it a try.